### PR TITLE
misc(deprecations): Add Tracing deprecations and replacements

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Enhancement

## :scroll: Description

This PR add deprecations and simple replacements which will be dropped in V6.

## :green_heart: How did you test it?
sample app, unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
 # Changelog
 
+## Unreleased
+
+### Changes
+
+- React Native Tracing Deprecations  ([#4073](https://github.com/getsentry/sentry-react-native/pull/4073))
```
Comment (lucas-zimerman) [inline CHANGELOG.md:7]: NIT:
```suggestion
- React Native Tracing Deprecations ([#4073](https://github.com/getsentry/sentry-react-native/pull/4073))
```

File: samples/react-native/src/App.tsx [hunk 2]
```diff
@@ -38,11 +38,10 @@ LogBox.ignoreAllLogs();
 
 const isMobileOs = Platform.OS === 'android' || Platform.OS === 'ios';
 
-const reactNavigationInstrumentation =
-  new Sentry.ReactNavigationInstrumentation({
-    routeChangeTimeoutMs: 500, // How long it will wait for the route change to complete. Default is 1000ms
-    enableTimeToInitialDisplay: isMobileOs,
-  });
+const reactNavigationInstrumentation = Sentry.reactNavigationIntegration({
+  routeChangeTimeoutMs: 500, // How long it will wait for the route change to complete. Default is 1000ms
```
Comment (lucas-zimerman) [inline samples/react-native/src/App.tsx:42]: Q: why using another value other than the default?

File: src/js/options.ts [hunk 3]
```diff
@@ -204,15 +204,52 @@ export interface BaseReactNativeOptions {
...
+
+  /**
+   * Track the app start time by adding measurements to the first route transaction. If there is no routing instrumentation
+   * an app start transaction will be started.
+   *
+   * Requires performance monitoring to be enabled.
+   *
+   * @default true
+   */
+  enableAppStartTracking?: boolean;
```
Comment (lucas-zimerman) [inline src/js/options.ts:221]: Q: Why nullable if we already set a default value as true, will null behave the same as setting it to false?

Same question applies to the below nullable boolean with null support.

File: test/client.test.ts [hunk 4]
```diff
@@ -688,6 +688,19 @@ describe('Tests ReactNativeClient', () => {
...
+    test('register user interactions tracing root option', () => {
+      const client = new ReactNativeClient(
+        mockedOptions({
+          enableUserInteractionTracing: true,
+          dsn: EXAMPLE_DSN,
+          integrations: [new ReactNativeTracing({})],
+        }),
+      );
+      client.setupIntegrations();
+
```
Comment (lucas-zimerman) [inline test/client.test.ts:700]:
```suggestion
```

File: src/js/options.ts (hunk 3 above)
Comment (krystofwoldrich) [reply src/js/options.ts:221]: To allow users to not specify the option in the init.

File: samples/react-native/src/App.tsx (hunk 2 above)
Comment (krystofwoldrich) [reply samples/react-native/src/App.tsx:42]: I have not updated the value here, just format. But the shorter value is here just for convenience when debugging.

**Pull request merged**